### PR TITLE
host predicates: add integration test for previous hosts

### PR DESCRIPTION
**Pull request by snowp**
Adds an integration test that demonstrates that the previous hosts retry host predicate works.

Risk Level: Low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Comment by snowp** [comment]
Adding since it was pointed out in #14735 that this doesn't have integration test coverage. cc @htuch

File: test/extensions/retry/host/previous_hosts/integration_test.cc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+namespace {
+
+class PrevioustHostsIntegrationTest : public testing::Test, public HttpIntegrationTest {
+public:
+  PrevioustHostsIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP2, Network::Address::IpVersion::v4) {}
+
+  void initialize() override {
+    // Add the retry configuration to a new virtual host.
+
```
Comment (alyssawilk) [inline test/extensions/retry/host/previous_hosts/integration_test.cc:21]: remove extra whitespace, and maybe move this comment down to the new vhost config?

**Review by alyssawilk — COMMENTED** [review]
Very nicely explained - thanks for the detailed comments.

LGTM modulo one minor nit

File: test/extensions/retry/host/previous_hosts/integration_test.cc (hunk 1 above)
Comment (alyssawilk) [reply test/extensions/retry/host/previous_hosts/integration_test.cc:20]: mind removing the newline before setDeterministic too?

**Review by alyssawilk — APPROVED** [review]
LGTM, pending happy CI :-)

**Pull request merged**